--- aufs/fs/aufs25/cpup.c~	2008-05-26 06:04:48.000000000 +0200
+++ aufs/fs/aufs25/cpup.c	2008-05-27 09:59:54.461647067 +0200
@@ -31,14 +31,7 @@
 static void au_fsstack_copy_inode_size(struct inode *inode,
 				       struct inode *h_inode)
 {
-	/* this spin_lock conflicts the new unionfs patch in -mm tree */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26) || !defined(MmTree)
-	spin_lock(&inode->i_lock);
 	fsstack_copy_inode_size(inode, h_inode);
-	spin_unlock(&inode->i_lock);
-#else
-	fsstack_copy_inode_size(inode, h_inode);
-#endif
 }
 
 void au_cpup_attr_timesizes(struct inode *inode)
